### PR TITLE
use ChaCha20Poly1305

### PR DESCRIPTION
**Pull request by iskakaushik**

### CODE REVIEW

File: flow/shared/enc.go [hunk 1]
```diff
@@ -34,6 +32,8 @@ func (e PeerDBEncKeys) Get(id string) (*PeerDBEncKey, error) {
 	return nil, fmt.Errorf("failed to find encryption key - %s", id)
 }
 
+const nonceSize = 12
```
Comment (serprex) [inline flow/shared/enc.go:35]: @heavycrystal do you think we need 24 for XChaCha?

Comment (heavycrystal) [reply flow/shared/enc.go:35]: hould use cipher.NewX and chacha20poly1305.NonceSizeX

Comment (serprex) [inline flow/shared/enc.go:35]:
```suggestion
const nonceSize = chacha20poly1305.NonceSizeX
```

Comment (heavycrystal) [reply flow/shared/enc.go:35]: granted XChaCha might be overkill

File: flow/shared/enc.go [hunk 2]
```diff
@@ -94,58 +84,16 @@ func (key *PeerDBEncKey) Encrypt(plaintext []byte) ([]byte, error) {
...
-		return nil, fmt.Errorf("failed to create new cipher: %w", err)
-	}
-
-	plaintext, err = PKCS7Pad(plaintext, aes.BlockSize)
-	if err != nil {
-		return nil, fmt.Errorf("failed to pad plaintext: %w", err)
-	}
-
-	ciphertext, err := safeAlloc(plaintext)
+	aead, err := chacha20poly1305.New(decodedKey)
```
Comment (serprex) [inline flow/shared/enc.go:87]:
```suggestion
	aead, err := chacha20poly1305.NewX(decodedKey)
```

File: nexus/catalog/Cargo.toml [hunk 3]
```diff
@@ -6,7 +6,6 @@ edition = "2021"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-aes = "0.8"
 anyhow = "1"
 async-trait = "0.1"
 base64 = "0.22"
```
Comment (serprex) [inline nexus/catalog/Cargo.toml:11]: also remove cbc crate

File: nexus/catalog/Cargo.toml [hunk 4]
```diff
@@ -28,3 +27,4 @@ tokio-postgres = { version = "0.7.6", features = [
 ] }
 tracing.workspace = true
 postgres-connection = { path = "../postgres-connection" }
+chacha20poly1305 = "0.10.1"
```
Comment (serprex) [inline nexus/catalog/Cargo.toml:30]: maintain alphabetical order

**Review by serprex — APPROVED** [review]

File: flow/shared/enc.go (hunk 1 above)
Comment (serprex) [reply flow/shared/enc.go:35]: overkill is okay

**Pull request merged**